### ISSUE TITLE
(Warped Mesh) Error On .nif Import Using Nifttools Addon v0.0.3 in Blender 2.83 and Blender 2.91

### ISSUE THREAD

**Issue by Vicarath** [niftools/blender_niftools_addon#413]
@niftools/blender-niftools-addon-reviewer -

Before creating a new issue, ensure that
 - [x] Read the user documentation.
 - [x] Check that the issue hasn't already been reported.
 - [x] Read through the list of `Helpful resources` links on the right-hand side.
 - [ ] Talk with the developers to ensure its a valid issue

The issue will get closed off immediately as invalid if
 - [ ] Looking for support where the forums, chat or documentation is more appropriate.
 - [ ] Goes off topic, loses focus or turns into a discussion.
 - [ ] Removing parts of the template the issue.

Fill out the template below to the best of your ability, including .blend files, nifs, logs; help us to help you.

# Issue Overview
I rediscovered The Elder Scrolls III Morrowind and wanted to try my hand at modding weapons into the game; upon searching for relevant tools to handle mesh packing and unpacking, 99% of resources found pointed me to NifSkope and the Niftools Blender Addon. I am more accustomed to Blender's control scheme and went in that direction. Although I believe I downloaded the correct file and followed the visual tutorial properly, almost every .nif I import causes an immediate error report in Blender, and the geometry for the mesh is warped almost beyond recognition.

### Blender Niftools Addon Version Info
Niftools Addon v0.0.3 from https://github.com/niftools/blender_niftools_addon/releases installed and activated according to the visual tutorial.

### Blender Version Info
Error occurred in Blender 2.83; however, updating to Blender 2.91 found the same error.

### Platform information
Edition: Windows 10 Home
Version: 20H2
OS Build: 19042.746

## Context
Only one .nif file has loaded without an error; every other file loads in with distorted geometry. The problem is, I want to edit an existing model, and I can't do that if the base geometry is a pile of rubbish.

## Steps to Reproduce
*I doubt this will be reproducible to the incredible devs behind this project, especially since neomonkeus verified the installation file only a few days ago.

1. I went to [this page](https://github.com/niftools/blender_niftools_addon/releases/tag/v0.0.3) and downloaded blender_niftools_addon-v0.0.3-2020-12-13-704c40d.zip

2. Following the [visual tutorial](https://user-images.githubusercontent.com/525919/98869622-fecdd380-2469-11eb-9605-946937de5923.gif) on that same page, I installed the addon from the zip file (without unzipping) from within Blender's preferences menu.

3. I imported a .nif I unpacked from Morrowind's .esm file using BSA Unpacker. Note that NifSkope opens the very same .nif with intact geometry (see side-by-side image under Screenshots).

4. I received the error message as detailed in the provided screenshot(s) and log.

While I did end up updating to Blender 2.91 and tried installing Pyffi as secondary measures, I ended up deleting the addon from within the Blender preferences menu, deleting the Pyffi and io_scene_Niftools folders (there weren't any other io_scene_X files in my addons folder or I would have deleted those, too), and going through the install process again. The result is the same error on .nif import.   

## Expected Result
I expected the .nif file to load in with its original mesh.

## Actual Result
The mesh loads in as a pile of rubbish.

### Info Bar Output
```
Dev: Sys variable not set
Executing - Niftools : Blender Niftools Addon v0.0.3(running on Blender 2.91.2, PyFFI 2.2.4.dev3)
Importing F:\Morrowind Unpacked BSAs\meshes\w\w_battleaxe_daedric.nif
NIF file version: 4000002
Reading file
Game set to 'FREEDOM_FORCE', but multiple games qualified
Importing data
Importing data for block 'W_battleaxe_daedric'
Importing data for block 'Tri W_battleaxe_daedric 0'
Importing mesh data for geometry 'Tri W_battleaxe_daedric 0'

Python: Traceback (most recent call last):
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\operators\nif_import_op.py", line 121, in execute
    return NifImport(self, context).execute()
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\nif_import.py", line 135, in execute
    self.import_root(root)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\nif_import.py", line 173, in import_root
    b_obj = self.import_branch(root_block)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\nif_import.py", line 254, in import_branch
    b_child = self.import_branch(n_child, b_armature=b_armature, n_armature=n_armature)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\nif_import.py", line 219, in import_branch
    return self.objecthelper.import_geometry_object(b_armature, n_block)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\modules\nif_import\object\__init__.py", line 142, in import_geometry_object
    self.mesh.import_mesh(n_block, b_obj)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\modules\nif_import\geometry\mesh\__init__.py", line 94, in import_mesh
    self.mesh_prop_processor.process_property_list(n_block, b_obj)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\modules\nif_import\property\geometry\mesh.py", line 113, in process_property_list
    self.process_property(prop)
  File "C:\Program Files\Blender Foundation\Blender 2.91\2.91\python\lib\functools.py", line 840, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\modules\nif_import\property\geometry\niproperty.py", line 126, in process_nitexturing_property
    NiTextureProp().import_nitextureprop_textures(prop, self._nodes_wrapper)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\io_scene_niftools\modules\nif_import\property\texture\types\nitextureprop.py", line 76, in import_nitextureprop_textures
    nodes_wrapper.create_and_link(slot_name, n_tex)
AttributeError: 'NoneType' object has no attribute 'create_and_link'

location: <unknown location>:-1
```

## Console Output:
![Full Log](https://user-images.githubusercontent.com/78367902/106487600-d3fb8600-6480-11eb-879c-c360cff8b89e.png)

## Attached .nif File:
[The Nif.zip](https://github.com/niftools/blender_niftools_addon/files/5905175/The.Nif.zip)

## Attached Texture File:
[Nif Texture.zip](https://github.com/niftools/blender_niftools_addon/files/5913871/Nif.Texture.zip)

## Screenshots:
![Side-By-Side](https://user-images.githubusercontent.com/78367902/106487660-e70e5600-6480-11eb-8a61-79ed6566ab01.png)
![NIF Error](https://user-images.githubusercontent.com/78367902/106487663-e7a6ec80-6480-11eb-89a8-7037c5b3d623.png)

## Similar Known Issues
While there are a few submitted issues that contain error messages/logs that seem similar to mine, none of the users behind these submissions completed the Bug report template, and they have the Failure or Incomplete tags as a result.

________________________________________________________________________________________________________________________________________

I appreciate any help with this! I'm still pretty much a noob to Blender, including addons, so it's entirely possible I missed something obvious like a checkbox I should have checked, etc. Also, if I messed up any of the instructions here, my apologies in advance; I'll do my best to rectify if this is the case. Love the work y'all do! I know there are tons of mods for this game that used this tool. :)

**Comment by neomonkeus** [comment]
The bug report is very thorough, thanks for taking the time to fill it out with the required detail.

Initially I suspect it is that being read as the wrong version type: `Game set to 'FREEDOM_FORCE', but multiple games qualified`

**Comment by neomonkeus** [comment]
Looks like an issue with loading the textures, did you unpack them as well?

**Comment by Vicarath** [comment]
Thanks for the response! I do have all the unpacked textures; they're in a separate folder, though... I just did a quick test where I put the matching texture in the same folder as the .nif, but importing still yields the same result on my end.

**Comment by neomonkeus** [comment]
No worries, I suspect what is happening is either the Texture node are is not being created, either due to input data or something different within tha specific nif.

I know it shouldn't make much difference but can you attach the corresponding texture files aswell.

**Comment by Vicarath** [comment]
Sure thing. I attached it to the main post, too.
[Nif Texture.zip](https://github.com/niftools/blender_niftools_addon/files/5913865/Nif.Texture.zip)

**Comment by neomonkeus** [comment]
@Vicarath Quick update, have a PR open under review which includes a fix for this issue. 
Should be merged quickly enough and will get v0.0.4 released in quick succession

**Comment by Vicarath** [comment]
Oh sweet! Thank you so much for working on it and keeping me updated!
